src/wp-includes/media.php

```diff
@@ -2117,6 +2117,13 @@ function wp_img_tag_add_width_and_height_attr( $image, $context, $attachment_id
 		$size_array = wp_image_src_get_dimensions( $image_src, $image_meta, $attachment_id );
 
 		if ( $size_array ) {
+			// If the width is enforced through style (e.g. in an inline image), calculate the dimension attributes.
+			$style_width = preg_match( '/style="width:\s*(\d+)px;"/', $image, $match_width ) ? (int) $match_width[1] : 0;
```

Comment:
Not sure if this is required or if GB inline images are the sole target, but this could be a bit more robust to handle cases like `style=" width:10px; "` or `style="color:red; width:10px;"`

```suggestion
			$style_width = preg_match( '/style="\s*(.*?;)?width:\s*(\d+)px;"/', $image, $match_width ) ? (int) $match_width[1] : 0;
```

Reply:
See my reply in https://github.com/WordPress/wordpress-develop/pull/5767#discussion_r1427176657. By making this more flexible to allow for any other style directives, it would open the possibility for a `height` also being present in which case the calculation logic would need to be expanded as well. Since inline images don't provide that, I'm not sure there's enough value to go for that flexibility.

We could always expand later if another use-case comes up that justifies it.